lib/matplotlib/rcsetup.py

```diff
@@ -242,13 +244,14 @@ def validate_fonttype(s):
 
 _validate_standard_backends = ValidateInStrings(
     'backend', all_backends, ignorecase=True)
+_auto_backend_sentinel = object()
```

Comment:
I think you may want to fix `rcParams.__getitem__("backend")` to check whether it would return the sentinel, and, if so, perform resolution and return an actual backend.  That seems less bad (especially now that backend resolution can be reverted) than returning something which has no `.lower()` method... (why are backends case insensitive?  that's an adventure for the next episode.)